src/pages/api/login.js

```diff
@@ -0,0 +1,96 @@
+import Boom from "@hapi/boom";
+import Joi from "@hapi/joi";
+import { verify } from "argon2";
+import cookie from "cookie";
+import { REFRESH_TOKEN_EXPIRES } from "../../../src/config";
+import { createErrorFor } from "../../../src/lib/apiError";
+import { getExpiryDate } from "../../../src/lib/duration";
+import { graphqlClient } from "../../../src/lib/graphqlClient";
+import { generateJwtToken } from "../../../src/lib/jwt";
+import { loginQuery, refreshTokenMutation } from "./login.gql";
+
+export default async function login(req, res) {
+  const apiError = createErrorFor(res);
+
+  if (req.method === "GET") {
+    res.setHeader("Allow", ["POST"]);
+    return apiError(Boom.methodNotAllowed("GET method not allowed"));
+  }
+  // validate username and password
+  const schema = Joi.object().keys({
+    username: Joi.string().required(),
+    password: Joi.string().required(),
+  });
+
+  const { error, value } = schema.validate(req.body);
+
+  if (error) {
+    console.error(error);
+    return apiError(Boom.badRequest(error.details[0].message));
+  }
+
+  const { username, password } = value;
+
+  let hasura_data;
+  try {
+    hasura_data = await graphqlClient.request(loginQuery, {
+      username,
```

Comment:
no password ?

Reply:
we verify the password after hashing it, but we could hash it first and pass to the request login. 
you prefer the first solution ?

Reply:
sounds more secure to make the comparison at hasura level

Reply:
In fact,  the library is made so you can't do that (hash and send to the DB), there is a `verify` method to do that

Reply:
ok right 👍